forte/processors/bert_based_query_creator.py

```diff
@@ -65,6 +71,21 @@ def get_embeddings(self, inputs, sequence_length, segment_ids):
 
         return cls_token
 
+    def _build_query(self, text: str) -> np.ndarray:
+        input_ids, segment_ids, input_mask = \
+            self.tokenizer.encode_text(
+                text_a=text, max_seq_length=self.config.max_seq_length)
+        input_ids = torch.LongTensor(input_ids).unsqueeze(0).to(self.device)
+        segment_ids = torch.LongTensor(segment_ids).unsqueeze(0).to(self.device)
+        input_mask = torch.LongTensor(input_mask).unsqueeze(0).to(self.device)
+        sequence_length = (1 - (input_mask == 0)).sum(dim=1)
+        query_vector = self.get_embeddings(inputs=input_ids,
+                                           sequence_length=sequence_length,
+                                           segment_ids=segment_ids)
+        query_vector = torch.mean(query_vector, dim=0, keepdim=True)
+        query_vector = query_vector.cpu().numpy()
+        return query_vector
+
     def _process(self, input_pack: MultiPack):
```

Comment:
Can this function in the abstract class?

Reply:
We could. But base class `QueryProcessor` is supposed to work with both `PackProcessor` and `MultiPackProcessor` and `_process` in `BertBasedQueryCreator` uses multi-pack related methods while accessing the data pack. Hence I thought it is better to define this method here.

Reply:
Huh, so we have hit the wall of multi-inherence so quick. That's something I am worrying about during the design phase.

Reply:
Added a `_process_query` method in `QueryProcessor`. Subclasses of `QueryProcessor` should now implement this method.